openflow_input/bsn_port_desc_module_eeprom

```diff
@@ -0,0 +1,228 @@
+    OFP_BSN_MODULE_EEPROM_IDENTIFIER_SFP = 0x3
+};
+
+enum ofp_bsn_module_eeprom_extended_identifier(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_EXT_IDENTIFIER_NON_GBIC = 0x00,
+    OFP_BSN_MODULE_EEPROM_EXT_IDENTIFIER_GBIC_SFP = 0x04,
+    OFP_BSN_MODULE_EEPROM_EXT_IDENTIFIER_GBIC_MOD_DEF = 0x07
+};
+
+enum ofp_bsn_module_eeprom_connector(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_NONE = 0x00,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_SC = 0x01,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FC_1_COPPER = 0x02,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FC_2_COPPER = 0x03,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_BNC_TNC = 0x04,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FC_COAX = 0x05,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_FIBERE_JACK = 0x06,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_LC = 0x07,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_MT_RJ = 0x08,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_MU = 0x09,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_SG = 0x0a,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_OPTICAL_PIGTAIL = 0x0b,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_MPO_PARALLEL_OPTIC = 0x0c,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_HSSDC = 0x20,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_COPPER_PIGTAIL = 0x21,
+    OFP_BSN_MODULE_EEPROM_CONNECTOR_RJ45 = 0x22
+};
+
+enum ofp_bsn_module_eeprom_encoding(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_ENCODING_UNSPECIFIED = 0x00,
+    OFP_BSN_MODULE_EEPROM_ENCODING_8B_10B = 0x01,
+    OFP_BSN_MODULE_EEPROM_ENCODING_4B_5B = 0x02,
+    OFP_BSN_MODULE_EEPROM_ENCODING_NRZ = 0x03,
+    OFP_BSN_MODULE_EEPROM_ENCODING_MANCHESTER = 0x04,
+    OFP_BSN_MODULE_EEPROM_ENCODING_SONET_SCRAMBLED = 0x05,
+    OFP_BSN_MODULE_EEPROM_ENCODING_64B_66B = 0x06
+};
+
+enum ofp_bsn_module_eeprom_rate_identifier(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_UNSPECIFIED = 0x00,
+    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_4G_2G_1G_AS0_AS1 = 0x01,
+    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_8G_4G_2G_RX_RATE_SELECT_ONLY = 0x02,
+    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_8G_4G_2G_INDEPENDENT_RX_TX = 0x03,
+    OFP_BSN_MODULE_EEPROM_RATE_IDENTIFIER_8G_4G_2G_TX_RATE_SELECT_ONLY = 0x04
+};
+
+enum ofp_bsn_module_eeprom_cu_cmplnce(wire_type=uint8_t, bitmask=False) {
+    OFP_BSN_MODULE_EEPROM_CU_CMPLLNCE_UNSPECIFIED = 0x00,
+    OFP_BSN_MODULE_EEPROM_CU_CMPLLNCE_SFF_8431_APP_E = 0x01,
+    OFP_BSN_MODULE_EEPROM_CU_CMPLLNCE_SFF_8431_LIMITING = 0x04
+};
+
+enum ofp_bsn_module_eeprom_alarm_set(wire_type=uint8_t, bitmask=True) {
+    OFP_BSN_MODULE_EEPROM_ALARM_NOT_SET = 0x00,
+    OFP_BSN_MODULE_EEPROM_ALARM_HI_SET = 0x01,
+    OFP_BSN_MODULE_EEPROM_ALARM_LOW_SET = 0x02,
+    OFP_BSN_MODULE_EEPROM_ALARM_HI_WARN_SET = 0x04,
+    OFP_BSN_MODULE_EEPROM_ALARM_LOW_WARN_SET = 0x08,
+};
+
+struct of_port_desc_prop_bsn_alarm : of_port_desc_prop_bsn {
+    uint16_t    type == 0xffff;
+    uint16_t    length;
+    uint32_t    experimenter == 0x5c16c7;
+    uint32_t    exp_type == 8;
```

Comment:
it looks like you are extending the of_port_desc_prop_bsn with more types, but there are no new ones defined on the extensions wiki page.  please update the wiki page with the ones you are adding.
also there does not appear to be an of_port_desc_prop_bsn with exp_type == 9 ; is this intentional ?

Reply:
Everything above 8 is commented out. Will need to remove them